src/main/kotlin/com/openlattice/rhizome/core/service/ContinuousRepeatingTaskService.kt

```diff
@@ -0,0 +1,152 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+
+import com.google.common.util.concurrent.ListeningExecutorService
+import com.hazelcast.core.HazelcastInstance
+import com.kryptnostic.rhizome.pods.HazelcastPod
+import org.slf4j.Logger
+import java.time.Instant
+import java.util.concurrent.Semaphore
+import java.util.concurrent.TimeUnit
+
+internal const val DEFAULT_BATCH_TIMEOUT_MILLIS = 120_000L
+
+abstract class ContinuousRepeatingTaskService<T: Any>(
+        private val executor: ListeningExecutorService,
+        hazelcastInstance: HazelcastInstance,
+        private val logger: Logger,
+        queueName: String,
+        lockingMapName: String,
+        parallelism: Int,
+        private val fillChunkSize: Int = 0,
+        private val batchTimeout: Long = DEFAULT_BATCH_TIMEOUT_MILLIS
+) {
+    private val locks = hazelcastInstance.getMap<T, Long>(lockingMapName)
+    private val candidates = hazelcastInstance.getQueue<T>(queueName)
+    private val limiter = Semaphore(parallelism)
+
+    private val enqueuer = if ( enqueuerEnabledCheck() ) {
+        executor.submit {
+            while (true) {
+                try {
+                    sourceSequence()
+                            .filter {
+                                val expiration = lockOrGetExpiration(it)
+                                HazelcastPod.logger.debug(
+                                        "Considering candidate {} with expiration {} at {}",
+                                        it,
+                                        expiration,
+                                        Instant.now().toEpochMilli()
+                                )
+                                if (expiration != null && Instant.now().toEpochMilli() >= expiration) {
+                                    HazelcastPod.logger.info("Refreshing expiration for {}", it)
+                                    //Assume original lock holder died, probably somewhat unsafe
+                                    refreshExpiration(it)
+                                    true
+                                } else expiration == null
+                            }.chunked(fillChunkSize)
+                            .forEach { keys ->
+                                candidates.addAll(keys)
+                                HazelcastPod.logger.info(
+                                        "Queued entities needing processing {}.",
+                                        keys
+                                )
+                            }
+                } catch (ex: Exception) {
+                    HazelcastPod.logger.info("Encountered error while enqueuing candidates for task.", ex)
```

Comment:
Can you make the logs a little bit more specific to the running task instead of just saying `task`?

Reply:
Yes, will do